### PR TITLE
again stuck at point 7 

### PR DESCRIPTION
**Pull request by roelof1967**
Hello, 

I had my test of point 4 working but on 6 adn 7 I had to change index.js and now if I run npm test I see this output : 

~~~
1)  Testing output on a valid url:                                                                                                                                                                   
                                                                                                                                                                                                       
      undefined is not a function                                                                                                                                                                      
                                                                                                                                                                                                       
      at /home/action/workspace/hueniversity/lib/index.js:25:91                                                                                                                                        
      at /home/action/workspace/hueniversity/node_modules/hapi/lib/server.js:180:24          
at Server.done (/home/action/workspace/hueniversity/node_modules/hapi/node_modules/items/lib/index.js:30:25)                                                                                     
      at Server.g (events.js:180:16)                                                                                                                                                                   
      at Server.EventEmitter.emit (events.js:117:20)                                                                                                                                                   
      at net.js:1055:10                                                                                                                                                                                
      at process._tickDomainCallback (node.js:459:13)
~~~

### CODE REVIEW

**Comment by EvanMarkHopkins** [comment]
I believe your issue is the data argument in your init function. When you call lab.init() in your test, you're unintentionally giving it your callback to 'data', when you meant it to be the second param.

**Comment by roelof1967** [comment]
Sorry , Can you explain it more into details. The only init in the test file is  this one lib.init(function(err, server
Or do you mean this line in index.js :  exports.init = function (data, callback)

**Comment by EvanMarkHopkins** [comment]
The init function you exported expects the arguments data & callback, but in your test you only provide one argument, which just so happens was supposed to be the second. Which has the opposite behavior of what I think you might have been counting on. By omitting the first argument, the callback took its place, inadvertently making the real callback (second argument) undefined.

**Comment by EvanMarkHopkins** [comment]
Sorry! ```lib``` unfortunate typo on my part

**Comment by EvanMarkHopkins** [comment]
Also, if you were to change 8000 to 0 in your index.js you could then provide 8000 as the first argument, and then any falsy value would make it default any available port. Just a thought.

**Comment by roelof1967** [comment]
Thanks, so the error is in this line :  lib.init(function(err, server) {

Sorry but im still confused

File: test/version.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+lab.test('Testing output on a valid url', function(done) {
+    
+    var options = {
+        method: 'GET',
+        url: '/version'
+    };
+    
+    lib.init(function(err, server) {
```
Comment (EvanMarkHopkins) [inline test/version.js]: Lib.init( ??? , function (err, server) { 
What is missing from here that is in your init function? You got to give it a port

Comment (roelof1967) [inline test/version.js]: Thanks, I got it working now.

**Comment by zoe-1** [comment]
@roelof1967  I began to try and help you.  
However, it seems that your git repo is a bit out of order. 

Do the following:
1.) Read the hueniversity instructions on working with git.
https://github.com/hueniverse/hueniversity/blob/master/guides/git.md
If you do not follow the instructions linked above your master branch work and future
assignments will be a mess. Everything written below is taken from that document except
where noted.

2.) Make a backup branch of your project state:
$ git branch your_backup_branch_name

3.) (Optional not from instructions above) 
In a different directory than your current hueniversity project,
make a clone of your git repo.  This will make a backup of your current work 
which you can copy and paste from later into branch assignment3.  
$ git clone https://github.com/roelof1967/hueniversity.git

3.) Replace your master branch with the current remote master branch.
$ git remote add upstream https://github.com/hueniverse/hueniversity
$ git fetch upstream
$ git checkout upstream/master -B master

4.) Push your updated master branch to your github repo ()
$ git push origin master --force

5.) Make a new branch called assignment3 and checkout that branch.
All code you write for assignment3 should be committed to this branch ('assignment3').
git branch assignment3
git checkout assignment3

6.) Write new code and when it is ready commit it to the assignment3 branch.
Before committing make sure you are on the right branch. 
$ git status  
$ git add .
$ git commit -m 'my changes'

7.) When you want others to read your assignment3 work just push the assignment3 
branch to github repo. In the future, your github repository will have multiple branches (master, assignment3, etc....).  
$ git push origin assignment3

If you follow the above workflow each assignment will have it's own branch which you can 
easily open and read when desired.  Plus, after each assignment you will be able to sync your
master branch to the hueniversity master branch without influencing your previous assignments work. 
Always follow the guide linked again below:
https://github.com/hueniverse/hueniversity/blob/master/guides/git.md

Hopefully, making the above corrections and restarting assignment3 will help you debug 
your current project.  At a minimum, these changes will help you and others access and reuse 
your work in the future. 
Regards.

**Pull request closed without merging**